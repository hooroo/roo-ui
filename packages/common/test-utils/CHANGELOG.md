# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
