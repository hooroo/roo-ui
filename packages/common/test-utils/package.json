{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
