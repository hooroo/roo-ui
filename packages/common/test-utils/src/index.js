export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
