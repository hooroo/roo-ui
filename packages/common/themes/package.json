{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
