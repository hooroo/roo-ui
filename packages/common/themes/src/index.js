import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
