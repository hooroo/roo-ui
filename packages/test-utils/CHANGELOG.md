# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.50.7"></a>
## [0.50.7](https://github.com/hooroo/roo-ui/compare/v0.50.6...v0.50.7) (2018-08-14)


### Bug Fixes

* pass theme to child components; update snapshots ([#153](https://github.com/hooroo/roo-ui/issues/153)) ([1a83477](https://github.com/hooroo/roo-ui/commit/1a83477))




<a name="0.39.1"></a>
## [0.39.1](https://github.com/hooroo/roo-ui/compare/v0.39.0...v0.39.1) (2018-06-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.39.0"></a>
# [0.39.0](https://github.com/hooroo/roo-ui/compare/v0.38.4...v0.39.0) (2018-06-26)


### Features

* add build:watch scripts ([#98](https://github.com/hooroo/roo-ui/issues/98)) ([bcae0bf](https://github.com/hooroo/roo-ui/commit/bcae0bf))




<a name="0.32.0"></a>
# [0.32.0](https://github.com/hooroo/roo-ui/compare/v0.31.0...v0.32.0) (2018-06-15)


### Features

* add autocomplete component ([#77](https://github.com/hooroo/roo-ui/issues/77)) ([eee3f8c](https://github.com/hooroo/roo-ui/commit/eee3f8c))




<a name="0.19.0"></a>
# [0.19.0](https://github.com/hooroo/roo-ui/compare/v0.18.0...v0.19.0) (2018-05-22)


### Features

* Add <List/> & <ListItem> ([#49](https://github.com/hooroo/roo-ui/issues/49)) ([b922a8a](https://github.com/hooroo/roo-ui/commit/b922a8a))




<a name="0.7.0"></a>
# [0.7.0](https://github.com/hooroo/roo-ui/compare/v0.6.4...v0.7.0) (2018-05-11)


### Features

* add new icon set ([#14](https://github.com/hooroo/roo-ui/issues/14)) ([377c4c8](https://github.com/hooroo/roo-ui/commit/377c4c8))




<a name="0.5.4"></a>
## [0.5.4](https://github.com/hooroo/roo-ui/compare/v0.5.2...v0.5.4) (2018-05-08)


### Bug Fixes

* replace build script with prepublish ([90cafe3](https://github.com/hooroo/roo-ui/commit/90cafe3))




<a name="0.5.3"></a>
## [0.5.3](https://github.com/hooroo/roo-ui/compare/v0.5.2...v0.5.3) (2018-05-08)


### Bug Fixes

* replace build script with prepublish ([90cafe3](https://github.com/hooroo/roo-ui/commit/90cafe3))




<a name="0.5.0"></a>
# [0.5.0](https://github.com/hooroo/roo-ui/compare/v0.4.2...v0.5.0) (2018-05-07)


### Features

* add ui primitives ([#13](https://github.com/hooroo/roo-ui/issues/13)) ([5789b96](https://github.com/hooroo/roo-ui/commit/5789b96))




<a name="0.4.2"></a>
## [0.4.2](https://github.com/hooroo/roo-ui/compare/v0.4.0...v0.4.2) (2018-04-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.4.1"></a>
## [0.4.1](https://github.com/hooroo/roo-ui/compare/v0.4.0...v0.4.1) (2018-04-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
