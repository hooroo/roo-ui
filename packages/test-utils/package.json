{
  "name": "@roo-ui/test-utils",
  "version": "0.5.3",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepublish": "babel src -d dist --copy-files"
  }
}
