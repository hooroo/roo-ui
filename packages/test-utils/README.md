# test-utils

## Installation

```shell
$ yarn add @roo-ui/test-utils
```

## Usage

### `shallowWithTheme`

Shallow render (with [Enzyme](http://airbnb.io/enzyme/)) a component along with theme context.

```js
import { shallowWithTheme } from '@roo-ui/test-utils';
import { qantas as theme } from '@roo-ui/themes';

import Example from '.';

describe('<Example />', () => {
  it('renders correctly', () => {
    expect(shallowWithTheme(<Example />, theme)).toMatchSnapshot;
  });
});
```
