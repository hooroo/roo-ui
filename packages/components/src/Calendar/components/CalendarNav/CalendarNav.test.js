import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import { Nav, NavButton } from '.';

describe('<Nav />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<Nav />, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<NavButton />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<NavButton>Next</NavButton>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
