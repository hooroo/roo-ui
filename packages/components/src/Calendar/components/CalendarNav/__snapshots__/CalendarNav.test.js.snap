// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarNav /> renders correctly 1`] = `
<Box>
  <NakedButton
    onClick={[MockFunction]}
  >
    <Icon
      name="chevronLeft"
    />
  </NakedButton>
  <NakedButton
    onClick={[MockFunction]}
  >
    <Icon
      name="chevronRight"
    />
  </NakedButton>
</Box>
`;
